### DISCUSSION TITLE
STM32F407VGT6 programming issues

### DISCUSSION THREAD

**Discussion by ITstreet1** [stm32duino/Arduino_Core_STM32#2302] [General]
Hello,

I designed a board based on the MCU from the tittle. Tried to upload a sketch using SWD and UART but no luck. No matter what I try, I get a message:

Error: failed to erase memory.

I tried to unlock the MCU over the STM32CubeProgrammer, but no luck either. It says it is set to AA. When I tried to erase the flash it said:
Error: Mass erase operation failed. Please verify flash protection.

However, when I use MikroElektronika CodeGrip and Necto Studio, I can easily program the chip. No issues here. 

Any idea what should I try to do? As far as I can see, I need to remove flash protection. How?

**Comment by fpistm** [comment]
You should check the Reference manual and see which option byte you should use. You can also read the https://www.st.com/resource/en/application_note/an2606-stm32-microcontroller-system-memory-boot-mode-stmicroelectronics.pdf, the first flash can require some specific operation.

**Reply by ITstreet1** [reply]
I flashed it over Necto, but still get the same in Arduino and STM32CubeProgrammer

**Reply by fpistm** [reply]
If you can't flash with cubeprogrammer no need to test with Arduino. As ide simply call the cubeprogrammer.

**Reply by ITstreet1** [reply]
Any idea how to unlock flash?

**Reply by fpistm** [reply]
Read the Reference manual and check the option bytes thanks CubeProgrammer.

**Reply by ITstreet1** [reply]
I am not sure I can follow you. Sorry about that.

Not sure what to look, and what to set in cube programmer

**Reply by ITstreet1** [reply]
However, I compiled the sketch in Arduino IDE and uploaded it to the MCU using CodeGrip and CodeGrip Suite. Zero issues.
Tried UART and Arduino IDE just to see if it settles somehow, but still nothing.

>       -------------------------------------------------------------------
>                        STM32CubeProgrammer v2.9.0                  
>       -------------------------------------------------------------------
> 
> Serial Port COM13 is successfully opened.
> Port configuration: parity = even, baudrate = 115200, data-bit = 8,
>                      stop-bit = 1.0, flow-control = off
> 
> Timeout error occured while waiting for acknowledgement.
> 
> Timeout error occured while waiting for acknowledgement.
> Error: Activating device: KO. Please, verify the boot mode configuration and check the serial port configuration. Reset your device then try again... 
> Error: Activating device: KO. Please, verify the boot mode configuration and check the serial port configuration. Reset your device then try again... 
> 
> 

Of course, it is in bootloader mode. When I upload a Blink sketch over the CodeGrip, the LED blinks. When I put it in the bootloader, it stops blinking. After resetting it blinks again.

**Reply by fpistm** [reply]
Well as stated while you can't achieve to flash using only CubeProgrammer no need to test Arduino IDE.
Your cubeProgrammer version seems old, you can try to update it.
Anyway, hard to help more on this as it is not linked to stm32 core issue.

**Reply by ITstreet1** [reply]
Updated cube programmer, the same.

Where should I check next? Can you suggest my next destination? 
https://www.stm32duino.com/
maybe...

**Reply by fpistm** [reply]
You can or stcommunity. Your issue is mainly using cube programmer with your b.oard

**Reply by ITstreet1** [reply]
It is interesting everything works just fine with MikroE tools...

**Comment by ITstreet1** [comment]
I started moving this project from the dead end.

First of all, this is my custom design. There ain't much soldered on the PCB. Only the MCU and what is necessary for it to work.
Now, while I can program this chip using MikroElektronika tools and IDE, I struggled with the Arduino IDE. The problem was two LEDs on the RX/TX lines from the STM32 to the CH340 UART chip. I have LEDs with 1K resistors, and when I removed them, it worked with the Arduino IDE. Now, should I use some other values of resistors, or just not add them to the design?

**Reply by fpistm** [reply]
Hi @ITstreet1 
Honestly I don't know as I'm not a hardware expert on board conception. I  advice to ask this on the ST community forum.
